xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp

```diff
@@ -662,34 +663,12 @@ std::string CDVDDemuxClient::GetFileName()
 
 std::string CDVDDemuxClient::GetStreamCodecName(int iStreamId)
 {
-  CDemuxStream *stream = GetStream(iStreamId);
-  std::string strName;
+  CDemuxStream* stream = GetStream(iStreamId);
   if (stream)
   {
-    if (stream->codec == AV_CODEC_ID_AC3)
-      strName = "ac3";
-    else if (stream->codec == AV_CODEC_ID_MP2)
-      strName = "mp2";
-    else if (stream->codec == AV_CODEC_ID_AAC)
-      strName = "aac";
-    else if (stream->codec == AV_CODEC_ID_DTS)
-      strName = "dca";
-    else if (stream->codec == AV_CODEC_ID_MPEG2VIDEO)
-      strName = "mpeg2video";
-    else if (stream->codec == AV_CODEC_ID_H264)
-      strName = "h264";
-    else if (stream->codec == AV_CODEC_ID_EAC3)
-      strName = "eac3";
-    else if (stream->codec == AV_CODEC_ID_VP8)
-      strName = "vp8";
-    else if (stream->codec == AV_CODEC_ID_VP9)
-      strName = "vp9";
-    else if (stream->codec == AV_CODEC_ID_HEVC)
-      strName = "hevc";
-    else if (stream->codec == AV_CODEC_ID_AV1)
-      strName = "av1";
```

Comment:
I was concerned as to why the video codec lines were being dropped from here as you don't appear to add them back anywhere else. However from testing it seems H.264/HEVC and VP9 for example are still showing in the GUI so I assume this is simply above my understanding.

Reply:
all these codec lines are standard ffmpeg values
since we can get these from ffmpeg by calling `avcodec_get_name`
there is no need to have them hardcoded

if you look at DVDDemuxFFmpeg.cpp
do the same thing so call `avcodec_get_name`

then i group the code under a common `GetCodecName` method
called here and from DVDDemuxFFmpeg.cpp